src/components/pages/donation_form/usePersonalDataSectionEventHandlers.ts

```diff
@@ -60,13 +60,16 @@ export function useAddressFormEventHandlers(
 
 		if ( !store.getters[ 'address/requiredFieldsAreValid' ] ) {
 			addressDataIsValid.value = false;
+			return;
 		}
 
 		if ( isDirectDebit.value && !store.getters[ 'bankdata/bankDataIsValid' ] ) {
 			bankDataIsValid.value = false;
+			return;
 		}
 
-		if ( !addressDataIsValid.value || !bankDataIsValid.value ) {
```

Comment:
This should be reverted to what it was. The reason we do the check after validating both the required fields and bankdata is that the summary will be removed if all the other fields apart from the IBAN become valid.

This is why the `shows and hides the error summary when payment data is invalid` test is failing.